contrib/test_integration/__init__.py

```diff
@@ -0,0 +1,155 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import filecmp
+import os
+import subprocess
+import sys
+import unittest
+
+sys.dont_write_bytecode = True  # prevent creation of .pyc files
+
+here = os.path.relpath(os.path.dirname(__file__))
+base_dir = os.path.join(here, "..", "..")
+merge_benchmark_sets = [
+    sys.executable,
+    os.path.join(base_dir, "contrib", "mergeBenchmarkSets.py"),
+]
+
+# Set to True to let tests overwrite the expected result with the actual result
+# instead of letting them fail.
+# Use this to update expected files if necessary. Do not commit this flag set to True!
+OVERWRITE_MODE = False
+
+
+class MergeBenchmarkSetsTests(unittest.TestCase):
```

Comment:
I would prefer to restructure the tests. `contrib/test_integration` sounds like it has tests for all of `contrib`, but if were to add tests for other components of `contrib`, we would want to keep them separate from the tests in this directory. Furthermore, a casual look at the `contrib` directory won't tell developers and users of `mergeBenchmarkSets.py` that there are tests for it. So better name the directory something like `test_mergeBenchmarkSets` and then structure the tests inside this according to the usual test structures with modules and classes.

Reply:
> and then structure the tests inside this according to the usual test structures with modules and classes.

I renamed the directory but am not sure what you mean by the above. Those tests are already structured like the other integration tests, no?

Reply:
I didn't specifically mean the other integration tests, just that the tests should have a good structure. Right now for example we have `test_mergeBenchmarkSets.TestXMLToString, test_mergeBenchmarkSets.TestWitnesses`, and `test_mergeBenchmarkSets.TestMerge` (all as classes in the same module) and `test_mergeBenchmarkSets_integration.MergeBenchmarkSetsIntegrationTests` (as a class in a package). Wouldn't it make more sense to have a package `test_mergeBenchmarkSets` and then structure all the tests appropriately within it using classes and modules as required?

Reply:
I see your point, but I deliberately separated unit- and integration tests like this to be consistent with other parts of benchexec.
I suppose having several classes in `test_mergeBenchmarkSets` is kind of arbitrary and makes it seem like there is not much of a connection between the tested functions, even though that isn't the case. My proposal would be to have only one class in `test_mergeBenchmarkSets` but otherwise leave the structure as-is.

Reply:
What is the argument against moving the integration tests into the package `test_mergeBenchmarkSets`, e.g., to a module like `test_mergeBenchmarkSets.integration`?

Reply:
Mainly consistency. It makes sense though, so I changed the test structure.